sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql

```diff
@@ -449,14 +462,7 @@ android_iap_subscriptions AS (
       "-",
       (plan_amount / 100)
     ) AS pricing_plan,
-    IF(
-      -- in billing grace period
-      auto_renewing
-      AND payment_state = 0,
-      -- grace period lasts from event_timestamp to end_time
-      end_time - event_timestamp,
-      INTERVAL 0 DAY
-    ) AS billing_grace_period,
+    IF(in_billing_grace_period, INTERVAL 1 MONTH, INTERVAL 0 DAY) AS billing_grace_period,
```

Comment:
Discussed this in person, and this is fine as is.

Reply:
Explanation:

`billing_grace_period` is *not* how long a sub has been in the grace period, it's how long a subscription's payment can be past due before service ends. For `Google Play` subs we don't actually know how long that is, instead we know whether a subscription is in it as of `event_timestamp` (and has not left before `DATE(CURRENT_TIMESTAMP)`), but we don't know how far into it, nor do we know when it will end. I thought we knew both of those, which is why I implemented it incorrectly before. We also know that `billing_grace_period` is no more than 1 month, and the subscription is updated after it ends to either be cancelled or paid.

`billing_grace_period` is used by `mozfun.norm.diff_months` to determine whether a user is far enough into a month to be considered "retained" for that month, so `in_billing_grace_perid = TRUE` is equivalent to `INTERVAL 1 MONTH` because the subscription is in the grace period now, and wasn't 1 month ago, regardless of when the last payment was due. I have tested that `mozfun.norm.diff_months` does correctly treat exactly one month as in the grace period with `INTERVAL 1 MONTH` even on otherwise weird month boundaries.